datadog_sync/model/logs_indexes_order.py

```diff
@@ -0,0 +1,93 @@
+# Unless explicitly stated otherwise all files in this repository are licensed
+# under the 3-clause BSD style license (see LICENSE).
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2019 Datadog, Inc.
+
+from __future__ import annotations
+from typing import TYPE_CHECKING, Optional, List, Dict, Tuple
+
+from datadog_sync.utils.base_resource import BaseResource, ResourceConfig
+from datadog_sync.utils.resource_utils import LogsIndexesOrderNameComparator
+
+if TYPE_CHECKING:
+    from datadog_sync.utils.custom_client import CustomClient
+
+
+class LogsIndexesOrder(BaseResource):
+    resource_type = "logs_indexes_order"
+    resource_config = ResourceConfig(
+        concurrent=False,
+        base_path="/api/v1/logs/config/index-order",
+        resource_connections={
+            "logs_indexes": ["index_names"],
+        },
+        deep_diff_config={
+            "ignore_order": False,
+            "custom_operators": [LogsIndexesOrderNameComparator()],
+        },
+    )
+    # Additional LogsIndexesOrder specific attributes
+    destination_indexes_order: Dict[str, Dict] = dict()
+    default_id: str = "logs-index-order"
+
+    async def get_resources(self, client: CustomClient) -> List[Dict]:
+        resp = await client.get(self.resource_config.base_path)
+
+        return [resp]
+
+    async def import_resource(self, _id: Optional[str] = None, resource: Optional[Dict] = None) -> Tuple[str, Dict]:
+        if _id:
+            source_client = self.config.source_client
+            resource = await source_client.get(self.resource_config.base_path)
+
+        return self.default_id, resource
+
+    async def pre_resource_action_hook(self, _id, resource: Dict) -> None:
+        pass
+
+    async def pre_apply_hook(self) -> None:
+        self.destination_indexes_order = await self.get_destination_indexes_order()
+
+    async def create_resource(self, _id: str, resource: Dict) -> Tuple[str, Dict]:
+        if not self.destination_indexes_order:
+            raise Exception("Failed to retrieve destination orgs logs index order")
+
+        self.resource_config.destination_resources[_id] = self.destination_indexes_order
+        return await self.update_resource(_id, resource)
+
+    async def update_resource(self, _id: str, resource: Dict) -> Tuple[str, Dict]:
+        destination_resources = self.destination_indexes_order or self.resource_config.destination_resources[_id]
+        ids_to_omit = set(resource["index_names"]) - set(destination_resources["index_names"])
+
+        extra_ids_to_include = [
+            _id for _id in destination_resources["index_names"] if _id not in resource["index_names"]
```

Comment:
nit: pick a different variable name here and below since `_id` is also a function parameter?